### ISSUE TITLE
Document MPI

### ISSUE THREAD

**Issue by mattwelborn** [MolSSI/QCEngine#221]
**Is your feature request related to a problem? Please describe.**
MPI is supported but not documented.

**Describe the solution you'd like**
There should be documentation for:

- running MPI with `compute`
- running MPI with `compute_procedure`
- developing MPI-aware program harnesses

**Comment by dgasmith** [comment]
@WardLT is this ready to be documented and supported?

**Comment by WardLT** [comment]
Yes, and I can make documentation for how the MPI-ready Program harnesses function. 

Where in the docs should it go? Programs or Developer Documentation?

**Comment by dgasmith** [comment]
I think a new page under "User Interface" since this is general to all program with a note in the environment detection page that MPI isn't automatically found.

**Comment by WardLT** [comment]
Thanks. I can do that
